app/controllers/answer_controller.rb

```diff
@@ -9,18 +9,9 @@ class AnswerController < ApplicationController
 
   def show
     @answer = Answer.includes(comments: %i[user smiles], question: [:user], smiles: [:user]).find(params[:id])
+    @subscribed = Subscription.where(user: current_user, answer: @answer).pluck(:id)
```

Comment:
the "subscribe" button never changes unless:

```suggestion
    @subscribed = Subscription.where(user: current_user, answer: @answer).pluck(:answer_id)
```

Reply:
as a guest this still makes a query, i'd rather have it not do one at all in that case.

something like this would be cool:
```ruby
@subscribed = []

...

if user_signed_in?
  @subscribed = Subscription.where ...
  mark_notifications_as_read
end
```